mkosi

```diff
@@ -675,8 +675,14 @@ def enable_networkd(workspace):
     os.remove(os.path.join(workspace, "root", "etc/resolv.conf"))
     os.symlink("../usr/lib/systemd/resolv.conf", os.path.join(workspace, "root", "etc/resolv.conf"))
 
-    patch_file(os.path.join(workspace, "root", "etc/nsswitch.conf"),
-               lambda line: " ".join(["resolve" if w == "dns" else w for w in line.split(" ")]) if line.startswith("hosts:") else line)
+    def fix_hosts_in_nsswitch(line):
+        if line.startswith("hosts:"):
+            sources = line.split(" ")
+            if 'resolve' not in sources:
+                return " ".join(["resolve" if w == "dns" else w for w in sources])
+        return line
```

Comment:
There's no reason for this to be an nested function. Please move it to the top level.

If there's nothing to do, it'd be nicer to just do nothing, instead of rewriting the file with identical contents.